packages/flutter/lib/src/material/ink_well.dart

```diff
@@ -608,8 +610,25 @@ class _InkResponseState<T extends InkResponse> extends State<T> with AutomaticKe
     return splash;
   }
 
+  void _handleFocusHighlightModeChange(FocusHighlightMode mode) {
+    if (!mounted) {
+      return;
+    }
+    setState(() {
+      _handleFocusUpdate();
+    });
+  }
+
   void _handleFocusUpdate() {
-    final bool showFocus = enabled && (Focus.of(context, nullOk: true)?.hasPrimaryFocus ?? false);
+    bool showFocus;
+    switch (WidgetsBinding.instance.focusManager.highlightMode) {
+      case FocusHighlightMode.touch:
+        showFocus = false;
```

Comment:
@gspencergoog Hi,
This change lets the touch phone disabled the `focusColor` without API documentations. I think we should improve the documentations.
https://github.com/flutter/flutter/blob/15d2d8a8750c2733871f7f8ae558c268efd32fc0/packages/flutter/lib/src/material/ink_well.dart#L439-L450

Sometimes it is useful that display the focus color, such as #70294 

Based on this, I want to add a property of  `InkWell` like `alwaysShowFocusColor`  to optional this behavior.
```dart
      case FocusHighlightMode.touch:
        showFocus = alwaysShowFocusColor  ? _shouldShowFocus : false;
        break;
```

Waiting for your thoughts! thanks very much.

Reply:
#70442